### PR TITLE
feat: display DropDown mentions suggestions in the expanded text field

### PR DESCRIPTION
**Pull request by ohassine**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When we expand the messaging text field, sending mentions will not work

### Causes (Optional)

Not implemented 

### Solutions

Display a DropdownMenu to show suggestions list of members to mention

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

- Expand  messaging textField
- Try to mention someone

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

https://user-images.githubusercontent.com/18124919/201720205-27392c7d-5c63-4619-9c35-fb843967547f.mp4

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1155/2/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/DropDownMentionsSuggestions.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        coordinateY = cursorCoordinateY + THIRTY
+    return coordinateY.toInt()
+}
+
+private const val MINIMUM_SCREEN_HEIGHT = 700
+private val DROP_DOWN_HEIGHT_SMALL = 150.dp
+private val DROP_DOWN_HEIGHT_LARGE = 200.dp
+private const val FIRST_LINE_INDEX = 0
+private const val HALF_SCREEN = 2.5
+private const val EIGHTY_PERCENT_OF_SCREEN = 0.80
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/DropDownMentionsSuggestions.kt]:
```suggestion
private const val EIGHTY_PERCENT = 0.80
```

**Review by vitorhugods — APPROVED** [review]
Everything looks pretty great to me.

Just one tiny naming suggestion and a thought:

image


It's a bit hard to realise that the dropdown is scrollable. Do you see a simple way of making the dropdown size about `2.75` or `3.25` times the size of an item? This way we're certain that one item will always be "peeking" and hint that the content is scrollable.

For example:
image

> **Note**: credits: https://css-tricks.com/thats-just-how-i-scroll/

**Comment by ohassine** [comment]
> Everything looks pretty great to me.
> 
> Just one tiny naming suggestion and a thought:
> 
> image
> 
> It's a bit hard to realise that the dropdown is scrollable. Do you see a simple way of making the dropdown size about `2.75` or `3.25` times the size of an item? This way we're certain that one item will always be "peeking" and hint that the content is scrollable.
> 
> For example: image
> 
> > Note: credits: https://css-tricks.com/thats-just-how-i-scroll/

Indeed, it's hard to know that there is more items to select.
I will try to make the last item partially visible as you suggested ✨

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1155/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1155/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4957026)(4957026)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4957026)(4957026)-dev-release.apk)

**Comment by gongracr** [comment]
What about maybe adding a slightly degraded translucent background on the end of the list @ohassine ?

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1155/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4962978)(4962978)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4962978)(4962978)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1155/8/) **failed**.

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-1155/9/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/DropDownMentionsSuggestions.kt [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    var sum = with(density) { DropdownMenuVerticalPadding.toPx().toInt() } * TWO
+    for ((i, itemSize) in itemHeights.toSortedMap()) {
+        sum += itemSize
+        if (sum >= baseHeightInt) {
+            return with(density) { (sum - itemSize / TWO).toDp() }
+        }
+    }
+    // all items fit into default height
+    return defaultMaxHeightDropdownMenu
+}
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/DropDownMentionsSuggestions.kt]:
```suggestion
private fun calculateMaxHeight(
    defaultMaxHeightDropdownMenu: Dp,
    itemHeights: Map<Int, Int>,
    density: Density,
    membersToMention: List<Contact>,
): Dp = if (itemHeights.keys.toSet() != membersToMention.indices.toSet()) {
        // if we don't have all heights calculated yet, return default value
        defaultMaxHeightDropdownMenu
    } else {
        val baseHeightInt = with(density) { defaultMaxHeightDropdownMenu.toPx().toInt() }
        var sum = with(density) { DropdownMenuVerticalPadding.toPx().toInt() } * TWO
        for ((i, itemSize) in itemHeights.toSortedMap()) {
            sum += itemSize
            if (sum >= baseHeightInt) {
                return with(density) { (sum - itemSize / TWO).toDp() }
            }
        }
        // all items fit into default height
        defaultMaxHeightDropdownMenu
}
```

**Review by vitorhugods — APPROVED** [review]
👨‍🍳🤌

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1155/10/) **failed**.

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-1155/11/) **failed**.

**Comment by AndroidBob** [comment]
Build [12](https://10.10.124.134/job/wire-android-reloaded/job/PR-1155/12/) **failed**.

**Comment by AndroidBob** [comment]
Build [13](https://10.10.124.134/job/wire-android-reloaded/job/PR-1155/13/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4971968)(4971968)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4971968)(4971968)-dev-release.apk)

**Pull request merged**